packages/botonic-plugin-text-classification/src/model/text-classifier.ts

```diff
@@ -1,14 +1,34 @@
+import { Preprocessor } from '@botonic/nlp/lib/preprocess/preprocessor'
 import { Intent } from '@botonic/nlp/lib/tasks/text-classification/process/prediction-processor'
-import { TextClassificationConfig } from '@botonic/nlp/lib/tasks/text-classification/storage/types'
+import type { TextClassificationConfig } from '@botonic/nlp/lib/tasks/text-classification/storage/types'
+import type { Tensor3D } from '@tensorflow/tfjs'
 import { LayersModel } from '@tensorflow/tfjs'
 
+import { PredictionProcessor } from '../process/prediction-processor'
+import { TextProcessor } from '../process/text-processor'
+
+// @ts-ignore
+global.fetch = fetch
```

Comment:
where do you use fetch?
Your link says it's a dirty hack we shoulnt use in production https://stackoverflow.com/a/51466112/145289
remove both usages if not required. if required, try to encapsulate in  a function or use a 3rd party package as suggested in  stackoverflow

Reply:
@dpinol , the reason of this is that `tfjs` uses fetch internally to load a model.json given a url, so the package will work correctly in browser envs but not in lambda (node). In the same thread there are some solutions like `node-fetch` or `cross-fetch`. We should give them a try :)

Reply:
@elozano98 can you create a ticket to solve this in a more elegant way?
Please add st like this:

> Your link says it's a dirty hack we shoulnt use in production https://stackoverflow.com/a/51466112/145289
> remove both usages if not required. if required, try to encapsulate in a function or use a 3rd party package as suggested in stackoverflow

Reply:
I have created [a ticket](https://linear.app/hubtype/issue/BOT-357/remove-fetch-from-nlp-plugins) for this (will be implemented in another PR).

Reply:
Fixed, cross-fetch used! @dpinol